### PR TITLE
Quest pokemon form filter

### PR DESCRIPTION
**Pull request by Karl-Parris**
## Description
Adds an option to show quest pokemon rewards by form name (eg Galarian, Alolan)

## Motivation and Context
Requested in #rocketmad-general. I would guess the main reasoning would be that people want to be able to find all galarian slowpokes in the current event (but they are flooded with regular ones)

## How Has This Been Tested?
Minimal impact. Just a frontend change that tweaks the functionality of the excludedQuestPokemon filter. Tested combinations of the pokemon filter and this new form filter and all appears to work as expected. Checked it correctly fills settings from store on load, doesn't affect anything when quests or quest filters are turned off.

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ReuschelCGN** [comment]
Tested but Filter didnt work for me. 
Any -> shows all Quest-Mons
Normal -> shows normal Quest-Mons
Galarian -> **dont** show Galarian Porenta (normal Mons are filtered out but galarian not shown)
Alola -> not testet

**Comment by Karl-Parris** [comment]
> Tested but Filter didnt work for me.
> Any -> shows all Quest-Mons
> Normal -> shows normal Quest-Mons
> Galarian -> **dont** show Galarian Porenta (normal Mons are filtered out but galarian not shown)
> Alola -> not testet

Ah, possibly a translation issue. Guessing your map isn't in English? "Normal" forms actually just have an empty form name, so I guess that would explain why "Normal" and "Any" works fine, but not "Galarian"/"Alola". I will take a look at solutions to this.

**Comment by ReuschelCGN** [comment]
Yes your right! It is german!

my map.html block looks: 

                      <select id="quest-form-filter">
                        <option value="Any">{{ i18n('Any') }}</option>
                        <option value="Galarian">{{ i18n('Galarian') }}</option>
                        <option value="Alola">{{ i18n('Alola') }}</option>
                        <option value="">{{ i18n('Normal') }}</option>
                      </select>

translation in de.json is right and shown. When I change option value to "Galar" it works as expected.

**Comment by ReuschelCGN** [comment]
```
<option value="Any">{{ i18n('Any') }}</option>
<option value="{{ i18n('Galarian') }}">{{ i18n('Galarian') }}</option>
<option value="{{ i18n('Alola') }}">{{ i18n('Alola') }}</option>
<option value="">{{ i18n('Normal') }}</option>
```
in map.html works with translation as expected

File: static/js/map/map.pokestop.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ function isPokestopMeetsQuestFilters(pokestop) {
                 return !settings.excludedQuestItems.has(id)
             }
             case 7: {
-                return !settings.excludedQuestPokemon.has(pokestop.quest.pokemon_id)
+                return !settings.excludedQuestPokemon.has(pokestop.quest.pokemon_id) &&
+                (settings.questFormFilter === 'Any' || settings.questFormFilter === getFormName(pokestop.quest.pokemon_id, pokestop.quest.form_id))
```
Comment (na-ji) [inline static/js/map/map.pokestop.js:29]: `getFormName` returns the translated form name, hence why you need to also translate the select values.

**Review by na-ji — CHANGES_REQUESTED** [review]
Thanks for the PR 👍

**Review by na-ji — APPROVED** [review]

**Pull request merged**